org.lflang/src/org/lflang/generator/LFGenerator.java

```diff
@@ -110,60 +112,73 @@ private GeneratorBase createGenerator(Target target, FileConfig fileConfig,
             case CCPP: {
                 return new CCppGenerator(fileConfig, errorReporter);
             }
-            case CPP: {
-                return createCppGenerator(fileConfig, errorReporter);
-            }
-            case TS: {
-                return new TypeScriptGenerator(
-                        (TypeScriptFileConfig) fileConfig, errorReporter);
-            }
             case Python: {
                 return new PythonGenerator(fileConfig, errorReporter);
             }
             default: {
-                throw new RuntimeException("Unexpected target!");
+                return createKotlinGenerator(target, fileConfig, errorReporter);
             }
         }
     }
 
     /**
-     * Create a C++ code generator
-     * 
-     * Since the CppGenerator class is implemented in Kotlin, the class is
+     * Create a code generator in Kotlin.
+     *
+     * Since the CppGenerator and TsGenerator class are implemented in Kotlin, the classes are
      * not visible from all contexts. If the RCA is run from within Eclipse via
      * "Run as Eclipse Application", the Kotlin classes are unfortunately not
      * available at runtime due to bugs in the Eclipse Kotlin plugin. (See
      * https://stackoverflow.com/questions/68095816/is-ist-possible-to-build-mixed-kotlin-and-java-applications-with-a-recent-eclips)
      * In this case, the method returns null
-     * 
-     * @return A CppGenerator object if the class can be found
+     *
+     * @return A Kotlin Generator object if the class can be found
      */
-    private GeneratorBase createCppGenerator(FileConfig fileConfig,
-            ErrorReporter errorReporter) {
+    private GeneratorBase createKotlinGenerator(Target target, FileConfig fileConfig,
+                                            ErrorReporter errorReporter) {
         // Since our Eclipse Plugin uses code injection via guice, we need to
-        // play a few tricks here so that CppFileConfig and CppGenerator do not
-        // appear as an import. Instead we look the class up at runtime and
-        // instantiate it if found.
+        // play a few tricks here so that Kotlin FileConfig and
+        // Kotlin Generator do not appear as an import. Instead we look the
+        // class up at runtime and instantiate it if found.
+        String packageName;
+        String classNamePrefix;
+        String targetName;
+        switch (target) {
+            case CPP: {
```

Comment:
If I'm not mistaken this info can be read straight from the `Target` enum. If not, it probably makes sense to store it there.

Reply:
I removed targetName, which is for the error messages.

I could toLowerCase and some string operations for package name and class name prefix; however, I think explicitly assigning the values makes the code more readable, and I'm afraid that for other languages, the relationship between enum and these names can be arbitrary.

Would this make sense to you?

Reply:
I think it's better to have fewer strings floating around and strive for uniformity. For the package name, `toLower` of the `name()` sounds appropriate. When it comes to the class name prefix, I guess the `toString()` could work? We have both `TypeScript` and `TS` floating around in the code base, and the Kotlin classes now use `Ts`; this is a perfect example of lack of uniformity that we can solve by adopting some conventions; the `Target` enum is the place to encode them.

Reply:
Got it. I did that. Thanks for the explanation.